### PR TITLE
Add a POSIX C LDK networking library 

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Turns out there is no good option in swift, so we should just do it in C here.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Updated to upstream + https://github.com/rust-bitcoin/rust-lightning/pull/988 (which hopefully will go in 0.0.99).

**Review by jkczyz — COMMENTED** [review]
Reviewed at a very high-level. Unable to run because `demo.cpp` needs to updated (see comment).

**Comment by TheBlueMatt** [comment]
Updated to latest upstream again.

**Review by jkczyz — APPROVED** [review]

File: lightning-c-bindings/Cargo.toml [hunk 1]
```diff
@@ -18,10 +18,10 @@ crate-type = ["staticlib"
...
-lightning = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "294009969aef617df4184ff41ef6daa5a445f213", features = ["allow_wallclock_use"] }
-lightning-persister = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "294009969aef617df4184ff41ef6daa5a445f213" }
-lightning-invoice = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "294009969aef617df4184ff41ef6daa5a445f213" }
-lightning-background-processor = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "294009969aef617df4184ff41ef6daa5a445f213" }
+lightning = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "865c6ba04eef70ed33afe484e9df7d12093fc598", features = ["allow_wallclock_use"] }
+lightning-persister = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "865c6ba04eef70ed33afe484e9df7d12093fc598" }
+lightning-invoice = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "865c6ba04eef70ed33afe484e9df7d12093fc598" }
+lightning-background-processor = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "865c6ba04eef70ed33afe484e9df7d12093fc598" }
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply lightning-c-bindings/Cargo.toml]: Huh! I'd intended to, but apparently set it to the wrong value. In any case I'll need to update it one more time before merge once 0.0.99 is tagged upstream, but that just changes the last commit so I won't bother with re-review.

**Pull request merged**